src/NServiceBus.AcceptanceTests/Sagas/When_using_ReplyToOriginator.cs

```diff
@@ -10,57 +10,83 @@
     public class When_using_ReplyToOriginator : NServiceBusAcceptanceTest
     {
         [Test]
-        public async Task Should_set_Reply_as_messageintent()
+        public async Task Should_preserve_correlation_context()
         {
+            var messageId = Guid.NewGuid().ToString();
+            var someId = Guid.NewGuid();
             var context = await Scenario.Define<Context>()
-                .WithEndpoint<Endpoint>(b => b.When(session => session.SendLocal(new InitiateRequestingSaga
-                {
-                    SomeCorrelationId = Guid.NewGuid()
-                })))
+                .WithEndpoint<Endpoint>(b => b.When(session =>
+                    {
+                        var option = new SendOptions();
+
+                        option.SetMessageId(messageId);
+                        option.RouteToThisEndpoint();
+
+                        return session.Send(new InitiateRequestingSaga
+                        {
+                            SomeId = someId
+                        }, option);
+                    })
+                    .When(c => c.GotFirstMessage, session =>
+                        //we're sending a new message here to make sure we get a new correlation id to make sure that the saga 
+                        //preserves the original one. If not the fact that we flow correlation ids from incoming messages to outgoing will
```

Comment:
the second sentence is hard to read. How about something like this: This message can't be sent from a message handler as it would float the received correlation id into the outgoing message instead of assigning a new one.